phoenix-core/src/main/java/org/apache/phoenix/compile/QueryCompiler.java

```diff
@@ -324,10 +328,13 @@ protected QueryPlan compileJoinQuery(StatementContext context, List<Object> bind
             QueryPlan plan = compileSingleFlatQuery(context, query, binds, asSubquery, !asSubquery && joinTable.isAllLeftJoin(), null, !table.isSubselect() && projectPKColumns ? tupleProjector : null, true);
             Expression postJoinFilterExpression = joinTable.compilePostFilterExpression(context, table);
             Integer limit = null;
+            Integer offset = null;
             if (!query.isAggregate() && !query.isDistinct() && query.getOrderBy().isEmpty()) {
                 limit = plan.getLimit();
+                offset = plan.getOffset();
             }
-            HashJoinInfo joinInfo = new HashJoinInfo(projectedTable, joinIds, joinExpressions, joinTypes, starJoinVector, tables, fieldPositions, postJoinFilterExpression, limit);
+            HashJoinInfo joinInfo = new HashJoinInfo(projectedTable, joinIds, joinExpressions, joinTypes,
+                    starJoinVector, tables, fieldPositions, postJoinFilterExpression, limit, offset);
```

Comment:
The "limit" in HashJoinInfo is a little different than the original SQL semantics. It specifies the max number of rows returned by a HashJoinRegionScanner so that it doesn't do too much unnecessary work, and its max rows per region, so in the end the client might get more rows than needed and will do a final limit operation (all done by ScanPlan, don't worry). So, here, I think the limit should be "offset + limit" and there is no need to add an extra "offset" field to HashJoinInfo. Btw, change in serialization/deserialization of HashJoinInfo would cause incompatibility between different versions.